src/cpp/processor/Processor.cpp

```diff
@@ -47,6 +47,25 @@ void Processor<EndPoint>::process_input_packet(
     {
         if (input_packet.message->prepare_next_submessage())
         {
+            switch (input_packet.message->get_subheader().submessage_id())
+            {
+                case dds::xrce::CREATE_CLIENT:
+                {
+                    process_create_client_submessage(input_packet);
+                    break;
+                }
+                case dds::xrce::GET_INFO:
+                {
+                    OutputPacket<EndPoint> output_packet;
+                    process_get_info_packet(std::move(input_packet), output_packet);
+                    server_.push_output_packet(std::move(output_packet));
+                    break;
+                }
+                default:
+                {
+                    break;
+                }
+            }
             if (input_packet.message->get_subheader().submessage_id() == dds::xrce::CREATE_CLIENT)
```

Comment:
This is done in the line above right? you are calling twice to `process_create_client_submessage`

Reply:
Fixed :man_facepalming: